File: 4. Matris/Python/Matrix.py
class Matrix:
    def __init__(self, row: int, column: int) -> None:
        self.__row = row
        self.__column = column
        self.__matrix = [[]]

    def set_matrix(self) -> None:
        print("Please enter the fields of the matrix according to the following pattern:\n\n1 2 3\n4 5 6\n7 8 9\n")
        i = 0
        while i < self.__row :
            input_str = input()
            j = 0
            row = []
            while j < self.__column :
                for entry in input_str :
                    if entry.isalnum() :
                        row.append(int(entry))
                        j += 1
            self.__matrix.append(row)
            i += 1
        del self.__matrix[0]

    def show(self) -> None:
        print("\n")
        for i in self.__matrix:
            print(i)

    def __set_zero(self, z_row: int, z_column: int) -> None:
        i = 0
        while i < self.__column:
            self.__matrix[z_row][i] = 0
            i += 1
        i = 0
        while i < self.__row:
            self.__matrix[i][z_column] = 0
            i += 1

    def zeroing(self) -> None:
        rows = []
        columns = []
        i = 0
        while i < self.__row:
            j = 0
            while j < self.__column:
                if self.__matrix[i][j] == 0 :
                    rows.append(i)
                    columns.append(j)
                j += 1
            i += 1
        
        i = 0
        while i < len(rows):
            self.__set_zero(rows[i], columns[i])
            i += 1

File: 4. Matris/C++/Matrix.cpp
#include "Matrix.h"

Matrix::Matrix(const int& row, const int& column) {
    this->row = row;
    this->column = column;
    matrix = new int*[row];
    for (int i = 0; i < row; i++)
        matrix[i] = new int[column];
}

Matrix::~Matrix() {
    for (int i = 0; i < row; i++)
        delete[] matrix[i];
    
    delete[] matrix;
}

void Matrix::set_matrix() {
    std::cout << "Please enter the fields of the matrix according to the following pattern:\n\n1 2 3\n4 5 6\n7 8 9\n" << std::endl;
    for (int i = 0; i < row; i++)
        for (int j = 0; j < column; j++)
            std::cin >> matrix[i][j];
    std::cout << "\nDone!" << std::endl;
}

void Matrix::show() {
    std::cout << std::endl;
    for (int i = 0; i < row; i++) {
        for (int j = 0; j < column; j++) {
            std::cout << matrix[i][j] << " ";
        }
        std::cout << std::endl;
    }
}

void Matrix::zeroing() {
    std::vector<int> rows;
    std::vector<int> columns;
    for (int i = 0; i < row; i++) {
        for (int j = 0; j < column; j++) {
            if (matrix[i][j] == 0) {
                rows.push_back(i);
                columns.push_back(j);
            }
        }
    }
    
    for (int i = 0; i < rows.size(); i++)
        set_zero(rows[i], columns[i]);
}